ui/apps/platform/src/Containers/Integrations/IntegrationsPage.tsx

```diff
@@ -15,16 +15,28 @@ import IntegrationsListPage from './IntegrationsListPage';
 import CreateIntegrationPage from './CreateIntegrationPage';
 import EditIntegrationPage from './EditIntegrationPage';
 import IntegrationDetailsPage from './IntegrationDetailsPage';
+import usePermissions from '../../hooks/usePermissions';
+import IntegrationsNoPermission from './IntegrationsNoPermission';
 
-const Page = (): ReactElement => (
-    <Switch>
-        <Route exact path={integrationsPath} component={IntegrationTilesPage} />
-        <Route exact path={integrationsListPath} component={IntegrationsListPage} />
-        <Route path={integrationCreatePath} component={CreateIntegrationPage} />
-        <Route path={integrationEditPath} component={EditIntegrationPage} />
-        <Route path={integrationDetailsPath} component={IntegrationDetailsPage} />
-        <Route component={IntegrationsNotFoundPage} />
-    </Switch>
-);
+const Page = (): ReactElement => {
+    const { hasReadAccess } = usePermissions();
+    const hasReadAccessForIntegrations = hasReadAccess('Integration');
+    return (
+        <>
+            {hasReadAccessForIntegrations ? (
+                <Switch>
+                    <Route exact path={integrationsPath} component={IntegrationTilesPage} />
+                    <Route exact path={integrationsListPath} component={IntegrationsListPage} />
+                    <Route path={integrationCreatePath} component={CreateIntegrationPage} />
+                    <Route path={integrationEditPath} component={EditIntegrationPage} />
+                    <Route path={integrationDetailsPath} component={IntegrationDetailsPage} />
+                    <Route component={IntegrationsNotFoundPage} />
+                </Switch>
+            ) : (
+                <IntegrationsNoPermission />
```

Comment:
What’s done is done. Was this a broken page which requires this change just before release?

Reply:
This is related to this discussion:
https://srox.slack.com/archives/C02L9F6KR25/p1667488912824189